src/Microsoft.IdentityModel.Clients.ActiveDirectory/Platforms/android/AndroidBrokerProxy.cs

```diff
@@ -226,21 +240,48 @@ public Intent GetIntentForBrokerActivity(AuthenticationRequest request, Activity
                 // logic
                 if (intent != null)
                 {
+                    _logger.Info("BrokerProxy: Intent created from BundleResult is not null. ");
                     intent.PutExtra(BrokerConstants.BrokerRequest, BrokerConstants.BrokerRequest);
                 }
+
+                _logger.Info("BrokerProxy: Intent created from BundleResult is null. ");
             }
-            catch (AdalException)
+            catch (AdalException ex)
             {
-                throw;
+                _logger.ErrorPii(ex);
+                throw new AdalException(
+                    AdalError.Unknown,
+                    ex.Message);
```

Comment:
why not re-throw?

Reply:
not following. isn't that what's happening?

---
In reply to: [286274862](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1598#discussion_r286274862) [](ancestors = 286274862)

Reply:
nope, a re-throw is throw. capturing the exception and throwing a new exception is .. that.

---
In reply to: [286733675](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1598#discussion_r286733675) [](ancestors = 286733675,286274862)

Reply:
got it. thanks. moved some things around. open to discussing what makes the most sense.

---
In reply to: [288389041](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1598#discussion_r288389041) [](ancestors = 288389041,286733675,286274862)